precompiles/oracle/src/lib.rs

```diff
@@ -0,0 +1,92 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use fp_evm::PrecompileHandle;
+use frame_support::dispatch::{GetDispatchInfo, PostDispatchInfo};
+use pallet_evm::AddressMapping;
+use pallet_oracle::TimestampedValue;
+use precompile_utils::{prelude::*, solidity};
+use sp_core::U256;
+use sp_runtime::traits::Dispatchable;
+use sp_std::marker::PhantomData;
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+pub struct OraclePrecompile<Runtime>(PhantomData<Runtime>);
+
+impl<Runtime> OraclePrecompile<Runtime>
+where
+    Runtime: pallet_oracle::Config<OracleKey = u32, OracleValue = u64> + pallet_evm::Config,
+    Runtime::RuntimeCall: Dispatchable<PostInfo = PostDispatchInfo> + GetDispatchInfo,
+    <Runtime::RuntimeCall as Dispatchable>::RuntimeOrigin: From<Option<Runtime::AccountId>>,
+    Runtime::RuntimeCall: From<pallet_oracle::Call<Runtime>>,
+{
+    pub fn new() -> Self {
+        Self(PhantomData)
+    }
+
+    /// Helper method to convert U256 to OracleKey
+    fn u256_to_key(key: U256) -> EvmResult<u32> {
+        Ok(key.low_u32())
+    }
+
+    /// Helper method to convert U256 to OracleValue
+    fn u256_to_value(value: U256) -> EvmResult<u64> {
+        Ok(value.low_u64())
+    }
+}
+
+#[precompile_utils::precompile]
+impl<Runtime> OraclePrecompile<Runtime>
+where
+    Runtime: pallet_oracle::Config<OracleKey = u32, OracleValue = u64> + pallet_evm::Config,
+    Runtime::RuntimeCall: Dispatchable<PostInfo = PostDispatchInfo> + GetDispatchInfo,
+    <Runtime::RuntimeCall as Dispatchable>::RuntimeOrigin: From<Option<Runtime::AccountId>>,
+    Runtime::RuntimeCall: From<pallet_oracle::Call<Runtime>>,
+{
+    #[precompile::public("feedValues(uint256[],uint256[])")]
+    fn feed_values(
+        handle: &mut impl PrecompileHandle,
+        keys: Vec<U256>,
+        values: Vec<U256>,
+    ) -> EvmResult {
+        handle.record_cost(RuntimeHelper::<Runtime>::db_write_gas_cost())?;
```

Comment:
This should be multiplied by the number of keys/values.
```suggestion
        handle.record_cost(RuntimeHelper::<Runtime>::db_write_gas_cost() * keys.len())?;
```